conan/tools/google/bazeldeps.py

```diff
@@ -0,0 +1,95 @@
+import textwrap
+
+from conans.util.files import save
+
+
+class BazelDeps(object):
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+
+    def generate(self):
+        local_repositories = []
+        for dependency in self._conanfile.dependencies.transitive_host_requires:
+            filename = self._create_bazel_buildfile(dependency)
+            local_repositories.append(self._create_new_local_repository(dependency, filename))
+
+        self._save_dependencies_file(local_repositories)
+
+    def _create_bazel_buildfile(self, dependency):
+        dependency.new_cpp_info.aggregate_components()
+
+        if not dependency.new_cpp_info.libs and not dependency.new_cpp_info.includedirs:
+            return None
+
+        result = 'load("@rules_cc//cc:defs.bzl", "cc_import", "cc_library")\n\n'
+
+        for lib in dependency.new_cpp_info.libs:
+            result += 'cc_import(\n'
+            result += '    name = "{0}_precompiled",\n'.format(lib)
+            result += '    static_library = "{0}/lib{1}.a"\n'.format(
+                dependency.new_cpp_info.libdirs[0],
+                lib
+            )
+            result += ')\n\n'
+
+        result += 'cc_library(\n'
+        result += '    name = "{}",\n'.format(dependency.ref.name)
+
+        if dependency.new_cpp_info.includedirs:
```

Comment:
This might read a bit easier with a jinja template

Reply:
Done